src/components/header/components/submenu-panel/index.js

```diff
@@ -61,6 +66,13 @@ class Header extends PureComponent {
     }
   };
 
+  handleClickOutside = (e) => {
```

Comment:
We are using a component called `<OutsideClickHandler />` that handles this for you keep the components clean. If you inspect the other menus this is already set at the root header component.